File: android/app/src/main/java/net/result/taulight/AndroidAgent.java
package net.result.taulight;

import net.result.sandnode.encryption.KeyStorageRegistry;
import net.result.taulight.hubagent.TauAgent;

import java.util.Map;

public class AndroidAgent extends TauAgent {
    private final Taulight taulight;
    private final String uuid;

    public AndroidAgent(KeyStorageRegistry keyStorageRegistry, Taulight taulight, String uuid) {
        super(keyStorageRegistry);
        this.taulight = taulight;
        this.uuid = uuid;
    }

    public void close() {
        taulight.sendToFlutter("disconnect", Map.of("uuid", uuid));
    }
}

File: android/app/src/main/java/net/result/taulight/AndroidAgent.kt
package net.result.taulight

import net.result.sandnode.config.AgentConfig
import net.result.sandnode.encryption.KeyStorageRegistry
import net.result.taulight.hubagent.TauAgent
import java.util.*

class AndroidAgent(val taulight: Taulight, val uuid: UUID, config: AgentConfig)
    : TauAgent(KeyStorageRegistry(), config) {

    override fun close() {
        taulight.sendToFlutter("disconnect", mapOf("uuid" to uuid.toString()))
    }
}
